osu.Game/Screens/Play/SpectatorPlayer.cs

```diff
@@ -68,6 +68,8 @@ protected override void LoadComplete()
                         master.UserPlaybackRate.Value = 1;
                 }
             }, true);
+
+            HUDOverlay.PlayerSettingsOverlay.PlaybackSettings.AllowControls.Value = false;
```

Comment:
Structurally I very much dislike that the thing is added at base `Player` level but then has to be disabled on `SpectatorPlayer`. If this could be added to `ReplayPlayer` directly, like the rest of the seek/pause gameplay logic is, this would be much cleaner.

I would like this PR to go in that direction. Yes `HUDOverlay` is non-overridable. Maybe it can be refactored easily to somehow be. Something like adding stuff to `HUDOverlay.PlayerSettingsOverlay` directly inside `ReplayPlayer`.

Reply:
I'm leaning on to refactoring it to allow derived players to implement their own `HUDOverlay`s as well since you have this weird oddity as well: https://github.com/ppy/osu/blob/master/osu.Game/Screens/Play/Player.cs#L480

Reply:
I'm not sure I'd agree. I'm not a fan of having to inherit multiple levels of classes just to change some details, especially so that `HUDOverlay` needs some key things to be hooked up to player correctly. I'd find it preferable if that could be avoided.